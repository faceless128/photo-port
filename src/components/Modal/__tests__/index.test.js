import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Modal from '..'

const currentPhoto = { name: "Cat green eyes", category: "portraits", description: "Portraits of people in my life", index: "1" };
const mockToggleModal = jest.fn();

afterEach(cleanup)

describe('Modal is rendering', () => {
  it('renders', () => {
    render(<Modal currentPhoto={currentPhoto} />);
  });
  
  it('renders', () => {
    const { asFragment } = render(<Modal currentPhoto={currentPhoto} />)
    expect(asFragment()).toMatchSnapshot()
  });
});

describe('Click Event', () => {
  it('calls onClose handler', () => {
    const { getByText } = render(<Modal
      onClose={mockToggleModal}
      currentPhoto={currentPhoto}
    />);
    fireEvent.click(getByText('Close this modal'));
  
    expect(mockToggleModal).toHaveBeenCalledTimes(1);
  });
})
