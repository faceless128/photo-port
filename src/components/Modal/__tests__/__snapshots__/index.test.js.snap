// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal is rendering renders 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        Cat green eyes
      </h3>
      <img
        alt="current category"
        src="1.jpg"
      />
      <p>
        Portraits of people in my life
      </p>
      <button
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
